src/hotspot/share/utilities/powerOfTwo.hpp

```diff
@@ -121,10 +121,17 @@ inline T next_power_of_2(T value)  {
 }
 
 // Find log2 value greater than this input
-template <typename T, ENABLE_IF(std::is_integral<T>::value)>
-inline T ceil_log2(T value) {
-  T ret;
-  for (ret = 1; ((T)1 << ret) < value; ++ret);
+template <typename T, typename U = typename std::make_unsigned<T>::type, ENABLE_IF(std::is_integral<T>::value)>
```

Comment:
U doesn't need to be a template parameter.  Make it a function-local type alias.